romancal/datamodels/tests/test_models.py

```diff
@@ -1,16 +1,186 @@
+            with pytest.warns(ValidationWarning) as record:
+                model.validate()
+        assert (len(record) > 0)
+
+# Testing core schema
+def test_core_schemas(tmp_path):
+
+    # Define meta dictionary of required key / value pairs
+    meta = {"meta":
+                {"instrument": {
+                     "detector": "WFI01",
+                     "optical_element": "F158",
+                     "name": "WFI"
+                 },
+                 "telescope": "ROMAN"
+                 }
+            }
+
+    # Test schema for required key/value pairs
+    confirm_required_keywords(tmp_path, meta)
+
+# Testing all reference file schemas
+def test_referencefile_model_schemas(tmp_path):
+
+    # Define base meta dictionary of required key / value pairs for reference files
+    # NOTE 1: date is commented out because it is not properly failing its test
+    #         Ticket: https://github.com/spacetelescope/stdatamodels/issues/23
+    # NOTE 2: core.schema key/value pairs commented out due to not properly failing their tests
+    #         Ticket: https://github.com/spacetelescope/stdatamodels/issues/7
+
+    meta = {"meta":
+                {
+                    "author": "Space Telescope Science Institute",
+                    "description": "Test file",
+                    "pedigree": "Test",
+                    "reftype": "REFERENCEFILE",
+                    "useafter": Time('1999-01-01T00:00:00.123456789',format='isot', scale='utc'),
+#                    "date": Time('1999-01-01T00:00:00.123456789',format='isot', scale='utc'),
+#                     # Key/value pairs from core.schema
+#                     "instrument": {
+#                         "detector": "WFI01",
+#                         "optical_element": "F158",
+#                         "name": "WFI"
+#                     },
+#                     "telescope": "ROMAN",
+                }
+    }
+
+    # Set temporary asdf file
+    file_path = tmp_path / "test.asdf"
+
+    # Cycle through reference file models to test each
+    for model_class in iter_subclasses(DataModels.referencefile.ReferenceFileModel):
+        # Test base referencefile model
+        if (model_class == DataModels.referencefile.ReferenceFileModel):
+            # Test required key/values pairs in meta dictionary
+            confirm_required_keywords(tmp_path, meta)
+
+            # Testing referencefile asdf file
+            with asdf.AsdfFile(meta) as af:
+                af.write_to(file_path)
+
+                # Load base referencefile schema
+                schema = asdf.schema.load_schema(
+                    "http://stsci.edu/schemas/roman_datamodel/reference_files/referencefile.schema")
+
+                # Ensure that base referencefile schema contains core schema
+                found = False
+                for schemaref in schema.get("allOf", [{}]):
+                    if schemaref.get("$ref","") == "../core.schema":
```

Comment:
Hmm... this seems like a good candidate for a assertion helper:

```python
def assert_referenced_schema(schema_uri, ref_uri):
  __tracebackhide__ = True
  schema = asdf.schema.load_schema(schema_uri)
  for subschema in schema.get("allOf", []):
    if asdf.generic_io.resolve_uri(schema_uri, subschema.get("$ref")) == ref_uri:
      return
  assert False, f"Schema '{schema_uri}' does not reference '{ref_uri}' in a top-level allOf combiner"
```
then we can use it here:
```python
assert_referenced_schema(ReferenceFileModel.schema_url, RomanDataModel.schema_url)
```
and also in the `FlatModel` test:
```python
assert_referenced_schema(FlatModel.schema_url, ReferenceFileModel.schema_url)
```

Reply:
Assertion helped.